lib/puppet_x/templates/init_ps.ps1

```diff
@@ -322,7 +322,41 @@ namespace Puppet
 }
 "@
 
-Add-Type -TypeDefinition $hostSource -Language CSharp
+# https://stackoverflow.com/questions/38790802/determine-operating-system-in-net-core
+# doesn't work
+# ([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform([System.Runtime.InteropServices.OSPlatform]::OSX))
+function Get-Platform
+{
+  $linuxOsTypePath = '/proc/sys/kernel/ostype'
+
+  if ($ENV:windir -and ($ENV:windir -contains '\\') -and (Test-Path $ENV:windir))
+  {
+    return 'Windows'
+  }
+  elseif (Test-Path $linuxOsTypePath)
+  {
+    $osType = Get-Content $osTypePath
+    if ($osType -imatch '^Linux') { return 'Linux' }     # Linux or Android
+
+    throw New-Object System.PlatformNotSupportedException($osType)
+  }
+  elseif (Test-Path '/System/Library/CoreServices/SystemVersion.plist')
+  {
+    return 'OSX' # iOS or Android
+  }
+
+  throw New-Object System.PlatformNotSupportedException
+}
+
+if ((Get-Platform) -eq 'OSX')
+{
+  # Works around Add-Type issue:
+  # Add-Type : The type initializer for 'Crypto' threw an exception
+  # Write-Host 'Detected OSX and changing DYLD_LIBRARY_PATH'
+  # TODO: works in current running session, but not clean session??
+  $Env:DYLD_LIBRARY_PATH = '/usr/local/opt/openssl/lib'
+}
+Add-Type -TypeDefinition $hostSource -Language CSharp -ReferencedAssemblies @('System.Collections', 'System.Console', 'System.Management.Automation', 'System.Globalization')
```

Comment:
This is currently problematic due to the bug at https://github.com/PowerShell/PowerShell/issues/1616

There has been some additional discussion on a few additional issues:

* https://github.com/dotnet/corefx/issues/5540
* https://github.com/dotnet/corefx/issues/12678
* https://github.com/dotnet/coreclr/issues/4651

I believe there needs to be a solution to the `ReferencedAssemblies` problem before this work can proceed, but it might be worth double checking against current releases on OSX - the release that I've been using is:

```
Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.10
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```